src/client/meshgen/extrusion_mesh_cache.cpp

```diff
@@ -0,0 +1,158 @@
+#include "extrusion_mesh_cache.h"
+#include "mesh.h"
+
+#define MIN_EXTRUSION_MESH_RESOLUTION 16
+#define MAX_EXTRUSION_MESH_RESOLUTION 512
+
+static scene::SMesh *createExtrusionMesh(int resolution_x, int resolution_y)
+{
+	const f32 r = 0.5;
+
+	scene::SMeshBuffer *buf = new scene::SMeshBuffer();
+	video::SColor c(255,255,255,255);
+	v3f scale(1.0, 1.0, 0.1);
+
+	// Front and back
+	{
+		video::S3DVertex vertices[8] = {
+			// z-
+			video::S3DVertex(-r,+r,-r, 0,0,-1, c, 0,0),
+			video::S3DVertex(+r,+r,-r, 0,0,-1, c, 1,0),
+			video::S3DVertex(+r,-r,-r, 0,0,-1, c, 1,1),
+			video::S3DVertex(-r,-r,-r, 0,0,-1, c, 0,1),
+			// z+
+			video::S3DVertex(-r,+r,+r, 0,0,+1, c, 0,0),
+			video::S3DVertex(-r,-r,+r, 0,0,+1, c, 0,1),
+			video::S3DVertex(+r,-r,+r, 0,0,+1, c, 1,1),
+			video::S3DVertex(+r,+r,+r, 0,0,+1, c, 1,0),
+		};
+		u16 indices[12] = {0,1,2,2,3,0,4,5,6,6,7,4};
+		buf->append(vertices, 8, indices, 12);
+	}
+
+	f32 pixelsize_x = 1 / (f32) resolution_x;
+	f32 pixelsize_y = 1 / (f32) resolution_y;
+
+	for (int i = 0; i < resolution_x; ++i) {
+		f32 pixelpos_x = i * pixelsize_x - 0.5;
+		f32 x0 = pixelpos_x;
+		f32 x1 = pixelpos_x + pixelsize_x;
+		f32 tex0 = (i + 0.1) * pixelsize_x;
+		f32 tex1 = (i + 0.9) * pixelsize_x;
+		video::S3DVertex vertices[8] = {
+			// x-
+			video::S3DVertex(x0,-r,-r, -1,0,0, c, tex0,1),
+			video::S3DVertex(x0,-r,+r, -1,0,0, c, tex1,1),
+			video::S3DVertex(x0,+r,+r, -1,0,0, c, tex1,0),
+			video::S3DVertex(x0,+r,-r, -1,0,0, c, tex0,0),
+			// x+
+			video::S3DVertex(x1,-r,-r, +1,0,0, c, tex0,1),
+			video::S3DVertex(x1,+r,-r, +1,0,0, c, tex0,0),
+			video::S3DVertex(x1,+r,+r, +1,0,0, c, tex1,0),
+			video::S3DVertex(x1,-r,+r, +1,0,0, c, tex1,1),
+		};
+		u16 indices[12] = {0,1,2,2,3,0,4,5,6,6,7,4};
+		buf->append(vertices, 8, indices, 12);
+	}
+	for (int i = 0; i < resolution_y; ++i) {
+		f32 pixelpos_y = i * pixelsize_y - 0.5;
+		f32 y0 = -pixelpos_y - pixelsize_y;
+		f32 y1 = -pixelpos_y;
+		f32 tex0 = (i + 0.1) * pixelsize_y;
+		f32 tex1 = (i + 0.9) * pixelsize_y;
+		video::S3DVertex vertices[8] = {
+			// y-
+			video::S3DVertex(-r,y0,-r, 0,-1,0, c, 0,tex0),
+			video::S3DVertex(+r,y0,-r, 0,-1,0, c, 1,tex0),
+			video::S3DVertex(+r,y0,+r, 0,-1,0, c, 1,tex1),
+			video::S3DVertex(-r,y0,+r, 0,-1,0, c, 0,tex1),
+			// y+
+			video::S3DVertex(-r,y1,-r, 0,+1,0, c, 0,tex0),
+			video::S3DVertex(-r,y1,+r, 0,+1,0, c, 0,tex1),
+			video::S3DVertex(+r,y1,+r, 0,+1,0, c, 1,tex1),
+			video::S3DVertex(+r,y1,-r, 0,+1,0, c, 1,tex0),
+		};
+		u16 indices[12] = {0,1,2,2,3,0,4,5,6,6,7,4};
```

Comment:
Couldn't this array be moved to the top of the function as static array? It's used multiple times in the loops.

Reply:
No, some values differ on each iteration.